implementation/observation-otel-bridge/src/main/java/io/smallrye/opentelemetry/instrumentation/observation/handler/AbstractTracingObservationHandler.java

```diff
@@ -108,16 +110,37 @@ protected Span getParentSpan(T context) {
         return null;
     }
 
+    @SuppressWarnings("unchecked")
     protected void tagSpan(T context, Span span) {
+        final Attributes highCardAttributes = context.get(HIGH_CARD_ATTRIBUTES);
+        setOtelAttributes(span, highCardAttributes);
+
+        final Attributes lowCardAttributes = context.get(LOW_CARD_ATTRIBUTES);
```

Comment:
Are these always set, no matter what?

Reply:
The low cardinality ones, yes.
The high cardinality are only used in traces.
This is there to go arround a limitation of the Observation API... Currently, It only supports storage of Strings. Will file a feature request.